


<launch>
  
  <node name="object_detector" pkg="dl_node" type="dl_node_sptam.py" output="screen" />
  
  <!-- Set use_sim_time true for datasets-->
  <param name="use_sim_time" value="true" />

  <!-- S-PTAM pose path publisher. Used for visualization. -->
  <node pkg="ros_utils" type="pose_to_path" name="sptam_path">
    <remap from="pose" to="sptam/robot/pose" />
    <remap from="path" to="sptam/robot/path" />
  </node>

  <!--
    Master nodelet manager.
    S-PTAM as well as the extraction nodelets will be run
    under a single manager to avoid unnecesary serialization of messages.
  -->
  <node name="nodelet_manager" pkg="nodelet" type="nodelet" args="manager" output="screen" clear_params="true" />

  <!-- S-PTAM node -->
  <node pkg="nodelet" type="nodelet" name="sptam" args="load sptam/sptam_nodelet nodelet_manager" output="screen" clear_params="true">

    <!-- Read S-PTAM parameters file -->
    <rosparam command="load" file="$(find sptam)/configurationFiles/kitti.yaml" />

    <param name="use_prediction" value="false" />
    <param name="approximate_sync" value="false" />
    <param name="publish_transform" value="true" />
    <param name="publish_on_fail" value="true" />

    <!--param name="prediction_frame" value="odom" /-->
    <param name="base_frame" value="left_camera" />
    <param name="camera_frame" value="left_camera" />
    <!--param name="map_frame" value="map" /-->
    <param name="reference_frame" value="left_camera" />

    <remap from="/stereo/left/image_rect" to="/stereo/left/image" />
    <remap from="/stereo/right/image_rect" to="/stereo/right/image" />
    <remap from="/stereo/left/camera_info"  to="/stereo/left/camera_info" />
    <remap from="/stereo/right/camera_info"  to="/stereo/right/camera_info" />

  </node>

</launch>
